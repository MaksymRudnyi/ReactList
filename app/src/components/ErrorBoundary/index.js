import React from 'react';
import PropTypes from 'prop-types';

export default class ErrorBoundary extends React.PureComponent {
  static propTypes = {
    children: PropTypes.any.isRequired,
  };

  state = {
    didCatch: false,
  };

  componentDidCatch() {
    this.setState({ didCatch: true });
  }

  render() {
    return this.state.didCatch ? null : this.props.children;
  }
}
