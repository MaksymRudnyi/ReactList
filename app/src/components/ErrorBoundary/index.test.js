import React from 'react';
import { mount } from 'enzyme';

import ErrorBoundary from '.';

describe('ErrorBoundary', () => {
  it('should render children', () => {
    expect(mount(<ErrorBoundary>content</ErrorBoundary>))
      .toMatchSnapshot();
  });

  it('should render error view when children throws error', () => {
    function BuggyComponent() {
      throw new Error();
    }

    expect(mount(<ErrorBoundary><BuggyComponent /></ErrorBoundary>))
      .toMatchSnapshot();
  });
});
