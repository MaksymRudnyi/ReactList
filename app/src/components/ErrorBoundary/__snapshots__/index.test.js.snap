// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary should render children 1`] = `ReactWrapper {}`;

exports[`ErrorBoundary should render error view when children throws error 1`] = `ReactWrapper {}`;
